src/test/java/pl/vm/javaguild/designpatterns/creational/factorymethodtest/FactoryMethodTest.java

```diff
@@ -0,0 +1,62 @@
+package pl.vm.javaguild.designpatterns.creational.factorymethodtest;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.Department;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.Enums.Section;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.Enums.Seniority;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.Enums.Stack;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.FinancialDepartment;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.ItDepartment;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Employee.Employee;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
```

Comment:
W jednym komentarzu zamieszczę uwagi dla całej klasy.

1. `SpringBootTest` jest tutaj nadmiarowe, bo nie potrzebujemy kontekstu springowego, czy dostępu do bazy.
2. Sugeruję następującą konstrukcję przy pisaniu testów: 
`
// given

// when

// then
`
Wiem, że może się wydawać nadmiarowa, ale poprawia czytelność i w mojej ocenie to dobra praktyka na przyszłość :)

3. Nie musisz się do tego teraz stosować, ale poczytaj sobie o testach parametrycznych z adnotacją `@ParametrizedTests` : [https://www.baeldung.com/parameterized-tests-junit-5](https://www.baeldung.com/parameterized-tests-junit-5). To jeden z ciekawszych ficzerów JUnit5. Niweluje logiczną duplikację kodu, przy podobnych testach, ale z różnymi parametrami.